tests/test_elocaps.py

```diff
@@ -0,0 +1,277 @@
+        headers={"Authorization": f"Bearer {users[0]['token']}"},
+    )
+    assert response1.status_code == 200 and len(response1.json()) == len(
+        [i for i in games if i.timestamp.date() == today]
+    )
+    assert response2.status_code == 200 and len(response2.json()) == len(
+        [i for i in games if i.timestamp.date() == yesterday]
+    )
+
+
+def test_get_game():
+    game = games[len(games) // 2]
+    response = client.get(
+        f"/elocaps/games/{game.id}",
+        headers={"Authorization": f"Bearer {users[0]['token']}"},
+    )
+    assert response.status_code == 200
+    json = response.json()
+    assert json["id"] == game.id
+    assert json["timestamp"] == game.timestamp.isoformat()
+    assert (
+        client.get(
+            "/elocaps/games/1234",
+            headers={"Authorization": f"Bearer {users[0]['token']}"},
+        ).status_code
+        == 404
+    )
+
+
+def test_validate_and_end_game():
+    game = games[0]
+    tokens = [
+        u["token"]
+        for i in game_players
+        if i.game_id == game.id
+        for u in users
+        if u["user"].id == i.user_id
+    ]
+    another_token = next(x for x in users if not x["token"] in tokens)["token"]
+    assert (
+        client.post(
+            f"/elocaps/games/{game.id}/validate",
+            headers={"Authorization": f"Bearer {another_token}"},
+        ).status_code
+        == 400
+    )
+    assert (
+        client.post(
+            f"/elocaps/games/{game.id}/validate",
+            headers={"Authorization": f"Bearer {tokens[0]}"},
+        ).status_code
+        == 201
+    )
+    assert (
+        client.post(
+            f"/elocaps/games/{game.id}/validate",
+            headers={"Authorization": f"Bearer {tokens[1]}"},
+        ).status_code
+        == 201
+    )
+    response = client.get(
+        f"/elocaps/games/{game.id}",
+        headers={"Authorization": f"Bearer {users[0]['token']}"},
+    )
+    assert (
+        response.status_code == 200
+        and response.json()["id"] == game.id
+        and response.json()["is_confirmed"]
+    )
+
+
+def test_get_waiting_games():
+    response = client.get(
+        "/elocaps/games/waiting",
+        headers={"Authorization": f"Bearer {users[0]['token']}"},
+    )
+    assert response.status_code == 200
+
+
+def test_player_games():
+    for i in range(4):
+        user = users[0]
+        response = client.get(
+            f"/elocaps/players/{user['user'].id}/games",
+            headers={"Authorization": f"Bearer {user['token']}"},
+        )
+        assert response.status_code == 200 and len(response.json()) == len(
+            [i for i in game_players if i.user_id == user["user"].id]
+        )
+
+
+def test_player_info():
+    for i in range(4):
+        response = client.get(
+            f"/elocaps/players/{users[i]['user'].id}",
+            headers={"Authorization": f"Bearer {users[i]['token']}"},
+        )
+        assert response.status_code == 200
```

Comment:
You should refactor this test to let pytest loop over the test
https://stackoverflow.com/questions/61679402/looping-over-tests-in-python-using-pytest